### PR TITLE
Database connection

### PR DESCRIPTION
**Pull request by tkolenbrander**
This pull request adds a databaseConnection class in order to connect with a MySQL database. The class has been tested with a line coverage of 97%.

### CODE REVIEW

**Comment by bvobart** [comment]
Your code looks good, nice and appropriate use of a Singleton there! PMD still gives some interesting warnings though: `ResultSet`s should be closed and take a look at `ConsecutiveAppendsShouldReuse` on [PMD's site about warnings about Strings](http://pmd.sourceforge.net/pmd-5.1.1/rules/java/strings.html)

File: src/main/java/nl/tudelft/contextproject/databaseConnection/DatabaseConnection.java [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+            int number = rs.getInt("number");
+            String shotId = rs.getString("shotId");
+            int cameraId = rs.getInt("camera");
+            int presetId = rs.getInt("preset");
+            String description = rs.getString("description");
+            String action = rs.getString("action");
+            Camera cam = Camera.getCamera(cameraId);
+            Preset preset = cam.getPreset(presetId);
+            Shot shot = new Shot(number, shotId, cam, preset, description, action);
+            script.addShot(shot);
```
Comment (stevenmeijer) [inline src/main/java/nl/tudelft/contextproject/databaseConnection/DatabaseConnection.java:227]: Might be more clear to group the different aspects and use some whilelines :)

**Comment by stevenmeijer** [comment]
Code looks nice, good work!

**Comment by tkolenbrander** [comment]
I implemented all the feedback.

**Comment by tkolenbrander** [comment]
@bvobart I won't fix the ConsecutiveAppendsShouldReuse, I think it would compromise the code readability.

**Comment by bvobart** [comment]
@tkolenbrander I don't see the problem. Use something like this:

```
StringBuilder sBuilder = new StringBuilder()
            .append("one")
            .append("two")
            .append("three");
```

**Comment by tkolenbrander** [comment]
@bvobart That is indeed a great solution, didn't think of that. Thank you. I will implement it.

**Comment by bvobart** [comment]
At your request, I've fixed almost all remaining PMD warnings and fixed the indentation with the `.append` calls. Once the build passes, this can be merged!

**Comment by stevenmeijer** [comment]
Looks good, will merge it when Travis passes!

**Pull request merged**